dao-services/src/main/java/org/exoplatform/wiki/jpa/search/AttachmentIndexingServiceConnector.java

```diff
@@ -87,16 +87,10 @@ public Document create(String id) {
     Map<String,String> fields = new HashMap<>();
     Document doc = new Document(TYPE, id, getUrl(attachment), fileItem.getFileInfo().getUpdatedDate(),
         computePermissions(fileItem.getFileInfo().getUpdater(),attachment), fields);
-    int index = fileItem.getFileInfo().getName().lastIndexOf(".");
-    if(index != -1) {
-      doc.addField("title", fileItem.getFileInfo().getName().substring(0, index - 1));
-    }
-    else{
-      doc.addField("title", fileItem.getFileInfo().getName());
-    }
-
+    String fileName = org.exoplatform.wiki.utils.Utils.escapeIllegalCharacterInName(fileItem.getFileInfo().getName());
+    doc.addField("title", fileName);
```

Comment:
Escaping special characters in the title results in having a difference between the real title and the indexed title. It should not IMO. Why can't we index title as is ?